hydrolib/core/io/structure/models.py

```diff
@@ -0,0 +1,129 @@
+from enum import Enum
+from hydrolib.core.io.ini.models import IniBasedModel
+from pathlib import Path
+from pydantic import validator, Field
+from typing import List, Literal, Optional, Union
+
+
+# TODO: handle comment blocks
+# TODO: handle duplicate keys
+class Structure(IniBasedModel):
+    # TODO: would we want to load this from something externally and generate these automatically
+    class Comments(IniBasedModel.Comments):
+        id: Optional[str] = "Unique structure id (max. 256 characters)."
+        name: Optional[str] = "Given name in the user interface."
+        branch_id: Optional[str] = Field(
+            "Branch on which the structure is located.", alias="branchId"
+        )
+        chainage: Optional[str] = "Chainage on the branch (m)."
+
+        n_coordinates: Optional[str] = Field(
+            "Number of values in xCoordinates and yCoordinates", alias="numCoordinates"
+        )
+        x_coordinates: Optional[str] = Field(
+            "x-coordinates of the location of the structure. (number of values = numCoordinates)",
+            alias="xCoordinates",
+        )
+        y_coordinates: Optional[str] = Field(
+            "y-coordinates of the location of the structure. (number of values = numCoordinates)",
+            alias="yCoordinates",
+        )
+
+    comments: Comments = Field(default_factory=Comments)
+
+    header: Literal["Structure"] = "Structure"
```

Comment:
This probably shouldn't be public?

Reply:
Probably a mistake in my configuration of the `IniBasedModel`, but when I make this private, pydantic does not parse the classes correctly any more.